src/main/java/ch/njol/skript/config/Node.java

```diff
@@ -214,12 +290,45 @@ protected String getIndentation() {
 	abstract String save_i();
 	
 	public final String save() {
-		return getIndentation() + save_i().replace("#", "##") + comment;
+		return getIndentation() + escapeUnquotedHashtags(save_i()) + comment;
 	}
 	
 	public void save(final PrintWriter w) {
 		w.println(save());
 	}
+
+	private static String escapeUnquotedHashtags(String input) {
+		int length = input.length();
+		StringBuilder output = new StringBuilder(input);
+		int numAdded = 0;
+		SplitLineState state = SplitLineState.CODE;
+		SplitLineState previousState = SplitLineState.CODE;
+		// find next " or %
+		for (int i = 0; i < length; i++) {
+			char c = input.charAt(i);
+			// check for things that can be escaped by doubling
+			if (c == '%' || c == '"' || c == '#') {
+				// escaped #s outside of strings
+				if (c == '#' && state != SplitLineState.STRING) {
+					output.insert(i + numAdded, "#");
+					numAdded++;
+					continue;
+				}
+				// skip if doubled (not #s)
+				if (i + 1 < length && input.charAt(i + 1) == c) {
+					i++;
+					continue;
+				}
+				SplitLineState tmp = state;
+				state = SplitLineState.update(c, state, previousState);
+				if (state == SplitLineState.HALT)
+					return output.toString();
```

Comment:
as discussed on discord, this would never be reached (as the input is always the code part of a line) and can be removed